portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,19 +34,28 @@
     
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-     //retrieves comment data from datastore
+     // Get the user input on how many comments to display
+     
+    Integer maxComments = getMaxComments(request);
+     //retrieves comment data from datastore 
+     //displays most recent comments first
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    //create arraylist of string to store comments
     ArrayList<String> comments = new ArrayList<String>();
+
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
+      //break when max comments are reached
+      if (comments.size() == maxComments){
```

Comment:
Same question as before, what if |comments.size()| is less than |maxComments|?

Reply:
You probably want to pass [`FetchOptions#limit`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html#limit-int-) to `results.asIterable` instead of fetching everything and then filtering on the app side.